inc/PassHash.class.php

```diff
@@ -27,6 +29,12 @@ function verify_hash($clear, $hash) {
         $magic  = '';
 
         //determine the used method and salt
+        if (substr($hash, 0, 2) == 'U$') {
+            // This may be an updated password from user_update_7000(). Such hashes
+            // have 'U' added as the first character and need an extra md5().
+            $hash = substr($hash, 1);
+            $clear = md5($clear);
+        }
```

Comment:
A minor thing, but less popular methods should be moved further down in the check (as long as it's possible, eg. before very broad checks that might match otherwise).

Reply:
Sure. However, this is does not easily fit in the long `if elseif elseif ...` list of test. The added `U` char does not identify the hash function. It only indicate to use `$clear=md5($clear)`. Very probably, when such a password is used, the hashing function will be Drupal's sha512. The problem is that I cannot be 100% sure of that. According to https://stackoverflow.com/a/5031807/1831273 the hash function should be Drupal's sha512. but this [Drupal 8 code](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Password%21PhpassHashedPassword.php/function/PhpassHashedPassword%3A%3Acheck/8.0.x) shows that Drupal can use other hashing functions. That's why this is done first, and the normal tests to determine the hashing function are run afterwards.

If you want, I can modify the test to check `U+sha512` and move it down.